### ISSUE TITLE
[ssr] Allow changing the default `location`

### ISSUE THREAD

**Issue by mohe2015** [lit/lit#2329]
# Description

To allow implementing a Router I think it would make sense that it is possible to specify the current url to the server-side `render` function. This should then change the value at https://github.com/lit/lit/blob/221cb0a90787631dcc867959de19febd2ebd3fd0/packages/labs/ssr/src/lib/dom-shim.ts#L134

**Comment by justinfagnani** [comment]
Yes :) We are working on a router ourselves right now and will need to implement this very soon,.

**Comment by kevinpschaaf** [comment]
For now, if you're using the `render-with-global-dom-shim.ts`, `window.location` is on the global and you can just set it before each call to `render`. This is what I've done in e.g. https://github.com/lit/lit/pull/1756/files#diff-ae7921011ddfcdaa30c0e29b37a1c01a9584e941e00186687c85f9a86ff6e6b9R40 to allow a client-side router to work.
